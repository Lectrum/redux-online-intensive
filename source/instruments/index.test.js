import { sum, delay } from './';

describe('instruments', () => {

    test('sum function should be a function', () => {
        expect(sum).toBeInstanceOf(Function);
    });

    test('function should throw exception if second argument has invalid type equals to string', () => {
        expect(() => sum(2, 'привет')).toThrow();
    });

    test('sum function must return an addition of two passed arguments', () => {
        expect(sum(2, 3)).toBe(5);
        expect(sum(1, 8)).toMatchSnapshot();
    });

    test('delay function should return a resolved promise', async () => {
        await expect(delay(5000)).resolves.toBeUndefined();
    });
});
