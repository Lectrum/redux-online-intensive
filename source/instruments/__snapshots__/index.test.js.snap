// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`instruments sum function must return an addition of two passed arguments 1`] = `9`;
